### DISCUSSION TITLE
DB file generated by TIA Portal to python dict

### DISCUSSION THREAD

**Discussion by Octopusmode** [gijzelaerr/python-snap7#333] [Ideas]
Hi, there
Is there a ready-to-use functionality to parse the TIA Portal generated .DB file (fig.1) to the layout specified in the example (fig.2)
To end up with a Python dictionary with variables indexed by type?

![image](https://user-images.githubusercontent.com/26312857/145483059-772121f4-0440-43f1-835b-5c594aa8db09.png)
Fig.1

![image](https://user-images.githubusercontent.com/26312857/145482995-bd96e15c-d34e-4bef-8731-4a105536c08d.png)
Fig.2

I'm trying to write a parser myself, but I'm not sure if my skills are enough.

**Comment by lautarodapin** [comment]
Use the `DB` class from the útiles module the same way as shown on that example that you mark
You can select the cells and columns name, Data type and offset, and copy directly to an excel, and then there change the column order.
Then copy it to a txt file or to a multiline python string 
Use tabs to separate it

**Comment by gijzelaerr** [comment]
Closing as answered. The DB class in snap7.util can be used to define data block layouts. TIA Portal column data can be exported to build the layout specification.
